### PR TITLE
Added possibility to exclude asset containers from optimization

### PR DESCRIPTION
**Pull request by kevinmeijer97**

### CODE REVIEW

**Comment by kevinmeijer97** [comment]
Quick note, created a story on the backlog to add some tests!

File: config/image-optimize.php [hunk 1]
```diff
@@ -15,4 +15,7 @@
 
     // The following mime types will be used to optimize images
     'mime_types' => ['image/jpeg', 'image/png', 'image/gif', 'image/webp'],
+
+    // You can exclude containers from optimization entirely here
+    'excluded_containers' => ['product_360_images'],
```
Comment (royduin) [inline config/image-optimize.php:20]: Don't think we want this value there by default 😇

**Review by royduin — CHANGES_REQUESTED** [review]
Config vs configurable in the cp? As containers can be created from the cp

**Comment by kevinmeijer97** [comment]
> Config vs configurable in the cp? As containers can be created from the cp

I figured to have it configurable through the config like the rest of the settings.
Although it might be a nice idea to add it as an extra setting on the asset container here:
Screenshot 2024-08-01 at 10 56 45

**Comment by royduin** [comment]
Nice improvement, maybe for later. Another thing; maybe it's nice to use a scope to have the conditions in one place: https://laravel.com/docs/master/eloquent#local-scopes. Currently it's duplicated on 3 places.

**Review by royduin — APPROVED** [review]

**Comment by kevinmeijer97** [comment]
> Nice improvement, maybe for later. Another thing; maybe it's nice to use a scope to have the conditions in one place: https://laravel.com/docs/master/eloquent#local-scopes. Currently it's duplicated on 3 places. Models do implement "macroable" so you could extend the model from the service provider. An example can be found here; https://laravel.com/docs/master/collections#extending-collections

Good idea, added a macro for this!

**Review by danielsmink — APPROVED** [review]

**Pull request merged**